src/runtime/zynq.cpp

```diff
@@ -35,6 +35,20 @@ extern int munmap(void *addr, size_t length);
 static int fd_hwacc = 0;
 static int fd_cma = 0;
 
+WEAK int halide_zynq_set_fd(int hwacc, int cma) {
+    if (!hwacc) {
```

Comment:
Do you mean to disallow calling halide_zynq_set_fd before the halide zynq runtime is initialized? I was think of the use case where the application can open the device and then call halide_zynq_set_fd and the halide function, and the halide_zynq_init in the halide function will be a no-op because the fd's are non-zeros.

Reply:
You are right. `halide_zynq_init` is not called in the halide function. I was confused with other Halide backends. I think I understand the code now.